src/TLuaInterpreter.cpp

```diff
@@ -15835,3 +15836,106 @@ int TLuaInterpreter::getMouseEvents(lua_State * L)
     }
     return 1;
 }
+
+int TLuaInterpreter::setMudletConfig(lua_State * L)
+{
+    QString key = getVerifiedString(L, __func__, 1, "key");
+    if (key.isEmpty()) {
+        return warnArgumentValue(L, __func__, "you must provide key");
+    }
+
+    auto success = [&]()
+    {
+        lua_pushboolean(L, true);
+        return 1;
+    };
+
+    Host& host = getHostFromLua(L);
+    if (host.mpMap && host.mpMap->mpMapper) {
+        if (key == "mapperRoomSize") {
+            host.mpMap->mpMapper->slot_setRoomSize(getVerifiedInt(L, __func__, 2, "value"));
+            return success();
+        }
+        if (key == "mapperExitSize") {
+            host.mpMap->mpMapper->slot_setExitSize(getVerifiedInt(L, __func__, 2, "value"));
+            return success();
+        }
+        if (key == "mapperRoundRooms") {
+            host.mpMap->mpMapper->slot_toggleRoundRooms(getVerifiedBool(L, __func__, 2, "value"));
+            return success();
+        }
+        if (key == "mapperShowRoomIds") {
+            host.mpMap->mpMapper->slot_setShowRoomIds(getVerifiedBool(L, __func__, 2, "value"));
+            return success();
+        }
+        if (key == "mapperShowMapInfo") {
```

Comment:
We already have a rich set of functions for this - https://wiki.mudlet.org/w/Manual:Lua_Functions#disableMapInfo and so on, don't need to duplicate it here. Maybe we can just mention that in the docs.

Reply:
Yeah, that's actually one of the problems i had - some of that stuff is already covered by other functions, some is not. I am pretty sure that once I start implementing more functionality I will run into two more issue
- there will be existing functions that allow setting something, but its functionality is somehow limited. Should I then duplicate it or not?
- there are probably methods that allow setting but not reading a preference - the only way to solve this will be to duplicate the functionality of the "setting" part

Therefore i think we need to have some methods duplicate. Simply an alternative way of setting the same thing. Internally they would use the same functions so the parameters are the same or similar

Reply:
@Mudlet/lua-interface ?

Reply:
No sure what you mean?

Reply:
Ah, just pulling in other Lua interface devs for an opinion.